Fix: correct search for control plane containers

alex123012 opened the pull request:
Now in functions `is_apiserver`, `is_etcd`, `is_controllermananager` and `is_scheduler` regexes for first command entry are not "strict", so when there is more than one container in control plane pod with first command entries, for example, `kube-apiserver` and `kube-apiserver-healthcheck` - checks for control plane containers would be applied also for another (non control plane) container. This produces false negative check results.

In this PR I proposed solution for fixing this behavior.

simar7 reviewed (dismissed):
could you give an example of how this would fail today (false positive)?

simar7 commented:
adding @chen-keinan to take a second look

alex123012 commented:
@simar7 Hi!
So, we are trying to run CIS benchmark with trivy-operator and receive fail for all `kube-apiserver` checks. Our `kube-apiserver` pod looks like:
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: kube-apiserver
  namespace: kube-system
spec:
  containers:
  - command:
    - kube-apiserver
    - "..."
    name: kube-apiserver
  - command:
    - /usr/local/bin/kube-apiserver-healthcheck
    name: healthcheck
```

With old regex it will check both containers in `kube-apiserver` pod:
<img width="1439" alt="image" src="https://user-images.githubusercontent.com/60808275/230561057-e16b779a-8bf7-4ee2-91fc-b92e6c78ee70.png">
(look at dev tools)

If we use regex from this PR - all works as expected (only `kube-apiserver` container is checked):
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/60808275/230561220-73026f1c-9f2e-4a98-ae46-2d6ca51e9924.png">

As you can see, examples differ only by regex in `is_apiserver` func

[playground](https://play.openpolicyagent.org/p/ajIpoTZ2US)

alex123012 commented:
@simar7 And hi here:)
I've added tests not to cis benchmark rules, but for lib itself.
Also i've fixed regex to proper validate abs paths for control plane binaries in pod command (refer to tests)

simar7 commented:
> @simar7 And hi here:)
> I've added tests not to cis benchmark rules, but for lib itself.
> Also i've fixed regex to proper validate abs paths for control plane binaries in pod command (refer to tests)

awesome, thanks!
